eng/build.sh

```diff
@@ -301,5 +301,16 @@ if [[ "$bootstrap" == true ]]; then
   bootstrap_dir=$_MakeBootstrapBuild
 fi
 
-BuildSolution
+if [[ "$restore" == true || "$build" == true || "$rebuild" == true || "$test_mono" == true ]]; then
+  BuildSolution
+fi
+
+if [[ "$test_core_clr" == true ]]; then
+  if [[ "$ci" == true ]]; then
+    runtests_args=""
+  else
+    runtests_args="--html"
+  fi
+  dotnet exec "$scriptroot/../artifacts/bin/RunTests/${configuration}/netcoreapp3.1/RunTests.dll" --tfm netcoreapp3.1 --tfm net5.0 --configuration ${configuration} --dotnet ${_InitializeDotNetCli}/dotnet $runtests_args
+fi
 ExitWithExitCode 0
```

Comment:
Have you guys consider rewriting the build script using C# ([possibly something like this](https://github.com/FakeItEasy/FakeItEasy/blob/master/tools/FakeItEasy.Build/Program.cs)). With top-level statements that would look the same too!

I think at this point it would make sense to push C# as the language of choice for this sort of things.

Reply:
Making it easier to deploy tooling like this is definitely on my radar. My personal goal is that we can get to a point in .NET 6 where we can `dotnet run some_file.cs`. Essentially have the `dotnet run` infrastructure generate project files in the background, compile them and then execute the resulting binary. At that point we can essentially have single C# file programs that we can run across all the OS that we support. That should make scenarios like this a lot easier to handle.

Reply:
If I never write another line of bash in my life, it will be too soon

...I say, before moving on to a task that will very likely involve writing bash

Reply:
> `dotnet run some_file.cs`

You will probably need to add some [nuget] references along the way. Personally I wouldn't mind having a minimal project file (which already has full tooling support), then [build.sh would reduce to this](https://github.com/FakeItEasy/FakeItEasy/blob/master/build.sh).